liferay-echo/liferay-portal
# LPS-197141 Cascade deletion for LayoutPageTemplateCollections

Motivation
=======
As now folders are available to organize the layout page templates and nesting is allowed, deletion must be modified as one folder is deleted, the ones nested within it should also be deleted, plus the DPTs as well.

Proposed Solution
========
Cascade deletion is implemented.

Steps to Verify
========

1. Go to Design > Page Templates > Display Page Templates
2. Create several nested folders containing also DPTs
3. Delete the root folder using the deletion from the management toolbar
4. Create several nested folders containing also DPT again
5. Delete the root folder using the kebab menu
6. Ensure in both cases no errors are thrown in the terminal and that the folders are deleted (e.g. create folders with the same names of the ones deleted, duplication is not allowed so if can create them again they where properly deleted)

Commits
========
- LPS-197141 Create new finder to retrieve layoutPageTemplateCollectionDescendantIds
- LPS-197141 buildService
- LPS-197141 Cascade deletion of LayoutPageTemplateCollections
